File: Python/Lexer.py
class Lexer:
    keywords = { "and" : AND,
                 "class" : CLASS,
                 "else" : ELSE,
                 "false" : FALSE,
                 "for" : FOR,
                 "fun" : FUN,
                 "if" : IF,
                 "nil" : NIL,
                 "or" : OR,
                 "print" : PRINT,
                 "return" : RETURN,
                 "super" : SUPER,
                 "this" : THIS,
                 "true" : TRUE,
                 "var" : VAR,
                 "while" : WHILE}
    start = 0
    current = 0
    line = 1
    tokens = []
    source = ""

    def __init__(self, source):
        self.source = source

    def scanTokens(self):
        while (!isAtEnd()):
            start = current
            scanToken();
        tokens.append(create_token(EOF, "", None, line))
        return tokens

    def scanToken(self):
        c = advance()

File: Python/Lexer.rs
/*
This isn't really meant to be a working lexer, I'm just playing around with how parts of a lexer
would be implemented in Rust if I were to write one in Rust.
*/

enum TokenType {

}

struct Lexer {
    source : String;
    tokens : Vec::new();
    start  : u64;
    current : u64;
    line    : u64;
}

fn scanToken(){
    c : char = advance();
    match(c){
        '(' => addToken(LEFT_PAREN);
        ')' => addToken(RIGHT_PAREN);
        '{' => addToken(LEFT_BRACE);
        '}' => addToken(RIGHT_BRACE);
        ',' => addToken(COMMA);
        '.' => addToken(DOT);
        '+' => addToken(PLUS);
        ';' => addToken(SEMICOLON);
        '*' => addToken(STAR);
        // add !, =, <, >, /, and their individual cases
        '"' => string();
        '\n' => line += 1;
    }
}
